common/main/java/com/couchbase/lite/internal/fleece/FLEncoder.java

```diff
@@ -335,6 +337,12 @@ public boolean write(@Nullable List<?> list) {
     @NonNull
     public byte[] finish() throws LiteCoreException { return impl.nFinish(getPeer()); }
 
+    // NOTE: the FLSliceResult returned by this method must be released by the caller
     @NonNull
     public FLSliceResult finish2() throws LiteCoreException { return impl.nFinish2(getPeer()); }
+
+    // NOTE: We expect the FLSliceResult returned by this method to be handed, immediately,
+    // to someone else (LiteCore) who will release it.  Java does release the memory.
+    @NonNull
+    public FLSliceResult finish3() throws LiteCoreException { return impl.nFinish3(getPeer()); }
```

Comment:
Trying to think about a better name than 1, 2, 3 but it's hard to come up with a good one. Maybe finishAndGetManagedFLSliceResult (how long is this?).